sizzlingstats/LogStats.cpp

```diff
@@ -275,6 +276,53 @@ void CLogStats::FireGameEvent( IGameEvent *event )
 		WriteLog( log );
 		//L 03/21/2011 - 02:35:56: "GooB<330><STEAM_0:1:23384772><Blue>" picked up item "tf_ammo_pack"
 	}
+	else if ( FStrEq( name, "arrow_impact" ) )
+	{
+		// Only count crusader's crossbow
+		if ( event->GetInt("projectileType") == 11 )
+		{
+		
+			int healer = event->GetInt( "shooter" );
+			playerInfo &pInfo1 = m_entIndexToPlayerInfo[healer];
+
+			int patient = event->GetInt( "attachedEntity" );
+			playerInfo &pInfo2 = m_entIndexToPlayerInfo[patient];
+
+
+			// Only count arrows for alive players and on the same team
+			if (pInfo1.pPlayerInfo->IsPlayer() && !pInfo1.pPlayerInfo->IsDead() 
+			 && pInfo2.pPlayerInfo->IsPlayer() && !pInfo2.pPlayerInfo->IsDead()
+			 && pInfo1.teamid == pInfo2.teamid)
+			{

```

Comment:
What if someone shoots an arrow and then quits the server before the arrow
connects?

On Mon, Aug 22, 2016 at 8:07 PM, Jordan Cristiano <notifications@github.com>
wrote:

> In sizzlingstats/LogStats.cpp
> <https://github.com/SizzlingStats/sizzlingplugins/pull/27#discussion_r75782581>
> :
>
> > +		// Only count crusader's crossbow
> > +		if ( event->GetInt("projectileType") == 11 )
> > +		{
> > +		
> > +			int healer = event->GetInt( "shooter" );
> > +			playerInfo &pInfo1 = m_entIndexToPlayerInfo[healer];
> > +
> > +			int patient = event->GetInt( "attachedEntity" );
> > +			playerInfo &pInfo2 = m_entIndexToPlayerInfo[patient];
> > +
> > +
> > +			// Only count arrows for alive players and on the same team
> > +			if (pInfo1.pPlayerInfo->IsPlayer() && !pInfo1.pPlayerInfo->IsDead()
> > +			 && pInfo2.pPlayerInfo->IsPlayer() && !pInfo2.pPlayerInfo->IsDead()
> > +			 && pInfo1.teamid == pInfo2.teamid)
> > +			{
>
> Thinking about it, it seems proper for players to have this stat counted
> regardless of whether they are alive or not. I believe the heal points from
> arrows are counted that way too. The IsPlayer check should be removed too.
> I forget the exact conditions for it to return true/false, but I don't
> think it's doing much for us here.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SizzlingStats/sizzlingplugins/pull/27/files/9ede4cb9ef2e78f32b5d4223c9b87a53d77203ba#r75782581>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACZ9ib2hMfy44RzzS4QNcinSp3czHwqjks5qijmogaJpZM4Jk3IH>
> .
>